src/test/java/com/redhat/pantheon/model/api/SlingResourceTest.java

```diff
@@ -0,0 +1,292 @@
+                        "jcr:date", Calendar.getInstance(),
+                        "jcr:number", "10")
+                .commit();
+
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        assertEquals("my-name", model.NAME.get());
+        assertEquals(new Long(10), model.NUMBER.get());
+        assertNotNull(model.DATE.get());
+    }
+
+    @Test
+    public void childMapping() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1")
+                .resource("child")
+                .resource("grandchild")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        // Then
+        model.CHILD
+                .get()
+                .GRANDCHILD
+                .get();
+    }
+
+    @Test
+    public void getDeepField() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1")
+                .resource("child")
+                .resource("grandchild",
+                        "jcr:name", "grandchild-name")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+
+        // Then
+        assertEquals("grandchild-name", model.GRANDCHILD_NAME.get());
+    }
+
+    @Test
+    public void fieldEditing() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:primaryType", "pant:module",
+                        "jcr:created", Calendar.getInstance(),
+                        "jcr:createdBy", "auser")
+                .commit();
+
+        // When
+        TestResource model = new TestResource(slingContext.resourceResolver().getResource("/content/module1"));
+        Calendar cal = Calendar.getInstance();
+        cal.set(Calendar.MONTH, 1);
+        cal.set(Calendar.DATE, 1);
+        cal.set(Calendar.YEAR, 2019);
+        model.DATE.set(cal);
+        model.NAME.set("someoneelse");
+        model.NUMBER.set(15L);
+
+        // Then
+        assertEquals(1, model.DATE.get().get(Calendar.MONTH));
+        assertEquals(1, model.DATE.get().get(Calendar.DATE));
+        assertEquals(2019, model.DATE.get().get(Calendar.YEAR));
+        assertEquals("someoneelse", model.NAME.get());
+        assertEquals(new Long(15), model.NUMBER.get());
+    }
+
+    @Test
+    public void testGetChild() throws Exception {
+        // Given
+        slingContext.build()
+                .resource("/content/module1",
+                        "jcr:primaryType", "pant:module",
+                        "jcr:created", Calendar.getInstance(),
+                        "jcr:createdBy", "auser")
+                .resource("child",
+                        "jcr:property", "prop-value")
```

Comment:
Based on this, it looks like `/content/module1` and `child` are totally separate resources. How does the framework know to tie them together as parent-child so that later your assertion passes?

Reply:
They are separate resources, and in this case line 109 is creating the child resource directly under the previously created one. That's how this api works... kind of weird. The framework then simply looks for the child resource under the parent one.

Reply:
Ahhhh, so the .resource.resource.resource syntax implies the relationship. I did not know that.